sdb/commands/stacks.py

```diff
@@ -338,39 +376,26 @@ def _call(self, objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
                 stacktrace_info += "{:<18s} {:<16s} {:6d}\n".format(
                     hex(tasks[0].value_()), task_state, len(tasks))
 
-            mod_match, func_match = False, False
-
-            #
-            # Note on the type-check being ignored:
-            # The original `stack_key` type is a list where the first
-            # element is a string and the rest of them are integers
-            # but this is not easily expressed in mypy, thus we ignore
-            # the assignment error below.
-            #
-            frame_pcs: List[int] = stack_key[1:]  #type: ignore[assignment]
+            frame_pcs: Tuple[int] = stack_key[1]
             for frame_pc in frame_pcs:
-                if mod_start != -1 and mod_start <= frame_pc < mod_end:
-                    mod_match = True
-
                 try:
                     sym = sdb.get_symbol(frame_pc)
-                    func, offset = sym.name, frame_pc - sym.address
-                    if self.args.function and self.args.function == func:
-                        func_match = True
+                    func = sym.name
+                    offset = frame_pc - sym.address
                 except LookupError:
-                    func, offset = hex(frame_pc), 0x0
-
-                #
-                # As a potential future item, we may want to print
-                # the frame with the module where the pc/function
-                # belongs to. For example:
-                #     txg_sync_thread+0x15e [zfs]
-                #
+                    func = hex(frame_pc)
+                    offset = 0x0
                 stacktrace_info += "{:18s}{}+{}\n".format("", func, hex(offset))
-
-            if mod_start != -1 and not mod_match:
-                continue
-            if self.args.function and not func_match:
-                continue
             print(stacktrace_info)
-        return []
+
+    def filter_threads(self,
+                       objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
+        return (task for task in objs if self.match_stack(task))
```

Comment:
I think you can use the built-in `filter()`. i.e. `filter(objs, self.match_stack)` instead of `filter_threads(objs)`.  This also has the advantage of not creating the list (it's a generator IIRC).

Reply:
Good idea, that's cleaner. Though I don't think I'm creating a list here - the `(...)` syntax is creating a generator as well.

Reply:
oh cool, I didn't realize thats what it did